File: IntranetBL/MovimientosTarjeta.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Data
Imports System.Data.SqlClient
Imports IntranetDA

''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
' DevelTec.mx 
' Clase Generada Utilizando el Generador de Codigo DeveltecCodeGenerator
''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
Public Class MovimientosTarjeta
    Private _connString As String = ""
    Public Sub New(ByVal connString As String)
        _connString = connString
    End Sub


    Public Function RecuperaCantidadMovimientosPorComprobar(ByVal id_empleado As Integer) As Integer
        Try
            Dim sql As String = "cantidad_movimientos_por_comprobar"

            Dim params As SqlParameter() = New SqlParameter(1) {}
            params(0) = New SqlParameter("id_empleado", id_empleado)

            Dim sqlHelp As New SqlHelper(_connString)

            Return sqlHelp.ExecuteScalar(sql, params)
        Catch ex As Exception
            Throw ex
        End Try
    End Function



    Public Function RecuperaMovimientosPorComprobar(ByVal id_empleado As Integer) As DataTable
        Try
            Dim sql As String = "recupera_movimientos_por_comprobar"

            Dim params As SqlParameter() = New SqlParameter(1) {}
            params(0) = New SqlParameter("id_empleado", id_empleado)

            Dim sqlHelp As New SqlHelper(_connString)

            Return sqlHelp.ExecuteDataAdapter(sql, params).Tables(0)
        Catch ex As Exception
            Throw ex
        End Try
    End Function



    Public Sub Guarda(ByVal tipo As String, ByVal numero_tarjeta As String, ByVal fecha_movimiento As DateTime,
            ByVal concepto As String, ByVal descripcion As String, ByVal tipo_comercio As String,
            ByVal moneda As String, ByVal num_autorizacion As String, ByVal estatus_tarjeta As String,
            ByVal tipo_cambio As Decimal, ByVal moneda_extranjera As Decimal, ByVal pesos As Decimal,
            ByVal id_usuario_registro As Integer, ByVal comision_iva As Decimal, ByVal archivo_origen As String, ByVal id_movimiento_tarjeta As String)

        Try
            Dim sql As String = "guarda_movimientos_tarjetas"

            Dim params As SqlParameter() = New SqlParameter(19) {}
            params(0) = New SqlParameter("tipo", tipo)
            params(1) = New SqlParameter("numero_tarjeta", numero_tarjeta)
            params(2) = New SqlParameter("fecha_movimiento", fecha_movimiento)
            params(3) = New SqlParameter("concepto", concepto)
            params(4) = New SqlParameter("descripcion", descripcion)
            params(5) = New SqlParameter("tipo_comercio", tipo_comercio)
            params(6) = New SqlParameter("moneda", moneda)
            params(7) = New SqlParameter("num_autorizacion", num_autorizacion)
            params(8) = New SqlParameter("estatus_tarjeta", estatus_tarjeta)
            params(9) = New SqlParameter("tipo_cambio", tipo_cambio)
            params(10) = New SqlParameter("moneda_extranjera", moneda_extranjera)
            params(11) = New SqlParameter("pesos", pesos)
            params(12) = New SqlParameter("id_usuario_registro", id_usuario_registro)

            params(13) = New SqlParameter("rfc_comercio", "")
            params(14) = New SqlParameter("no_control_edo_cuenta", "")
            params(15) = New SqlParameter("abono", 0)
            params(16) = New SqlParameter("comision_iva", comision_iva)
            params(17) = New SqlParameter("archivo_origen", archivo_origen)
            params(18) = New SqlParameter("id_movimiento_tarjeta", id_movimiento_tarjeta)


            Dim sqlHelp As New SqlHelper(_connString)

            sqlHelp.ExecuteNonQuery(sql, params)
        Catch ex As Exception
            Throw ex
        End Try
    End Sub


    Public Function AgregarSolicitudComprobate(ByVal id_solicitud As Integer, ByVal tipo As String, ByVal id_movimiento As Integer, ByVal id_concepto As Integer,
                                          ByVal tipo_comprobantes As String, ByVal tipo_comprobacion_valor As String, ByVal necesidad As String, ByVal iva As Decimal,
                                          ByVal otros_impuestos As Decimal, ByVal retencion As Decimal, ByVal num_personas As Integer, ByVal iva_manual As Decimal,
                                          ByVal propina As Decimal, ByVal observaciones As String) As Integer

        Dim respuesta As Integer = 0
        Try
            Dim sql As String = "guarda_comprobante_movimientos_tarjetas_en_solicitud"

            Dim params As SqlParameter() = New SqlParameter(14) {}
            params(0) = New SqlParameter("id_solicitud", id_solicitud)
            params(1) = New SqlParameter("tipo", tipo)
            params(2) = New SqlParameter("id_movimiento", id_movimiento)
            params(3) = New SqlParameter("id_concepto", id_concepto)
            params(4) = New SqlParameter("tipo_comprobantes", tipo_comprobantes)
            params(5) = New SqlParameter("tipo_comprobacion_valor", tipo_comprobacion_valor)
            params(6) = New SqlParameter("necesidad", necesidad)
            params(7) = New SqlParameter("iva", iva)
            params(8) = New SqlParameter("otros_impuestos", otros_impuestos)
            params(9) = New SqlParameter("retencion", retencion)
            params(10) = New SqlParameter("num_personas", num_personas)
            params(11) = New SqlParameter("iva_manual", iva_manual)
            params(12) = New SqlParameter("propina", propina)
            params(13) = New SqlParameter("observaciones_extra", observaciones)

            Dim sqlHelp As New SqlHelper(_connString)

            respuesta = sqlHelp.ExecuteScalar(sql, params)
        Catch ex As Exception
            Throw ex
        End Try

        Return respuesta
    End Function


End Class

File: ImportMovimientosEDENRED/MovimientosTarjeta.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
//using System.Threading.Tasks;

namespace ImportMovimientosEDENRED
{
    class MovimientosTarjeta
    {
        string _connString = "";

        public MovimientosTarjeta(string connString)
        {
            _connString = connString;
        }

        public void Guarda(string tipo, DateTime fecha_movimiento, string numero_tarjeta, string estatus_tarjeta, string concepto, string descripcion, string tipo_comercio, string rfc_comercio, string id_movimiento, string num_autorizacion, string no_control_edo_cuenta, decimal abono, decimal valor_pesos, decimal comision_iva, decimal valor_moneda_extranjera, string desc_moneda_extranjera, decimal tipo_cambio, int id_usuario_registro, string archivo_origen)
        {

            try
            {
                string sql = "guarda_movimientos_tarjetas";

                SqlParameter[] @params = new SqlParameter[19];
                @params[0] = new SqlParameter("tipo", tipo);
                @params[1] = new SqlParameter("fecha_movimiento", fecha_movimiento);
                @params[2] = new SqlParameter("numero_tarjeta", numero_tarjeta);
                @params[3] = new SqlParameter("estatus_tarjeta", estatus_tarjeta);
                @params[4] = new SqlParameter("concepto", concepto);
                @params[5] = new SqlParameter("descripcion", descripcion);
                @params[6] = new SqlParameter("tipo_comercio", tipo_comercio);
                @params[7] = new SqlParameter("rfc_comercio", rfc_comercio);
                @params[8] = new SqlParameter("id_movimiento_tarjeta", id_movimiento);
                @params[9] = new SqlParameter("num_autorizacion", num_autorizacion);
                @params[10] = new SqlParameter("no_control_edo_cuenta", no_control_edo_cuenta);
                @params[11] = new SqlParameter("abono", abono);
                @params[12] = new SqlParameter("pesos", valor_pesos);
                @params[13] = new SqlParameter("comision_iva", comision_iva);
                @params[14] = new SqlParameter("moneda_extranjera", valor_moneda_extranjera);
                @params[15] = new SqlParameter("moneda", desc_moneda_extranjera);
                @params[16] = new SqlParameter("tipo_cambio", tipo_cambio);
                @params[17] = new SqlParameter("id_usuario_registro", id_usuario_registro);
                @params[18] = new SqlParameter("archivo_origen", archivo_origen);


                SqlHelper sqlHelp = new SqlHelper(_connString);

                sqlHelp.ExecuteNonQuery(sql, @params);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }


        public void GuardaDatosArchivo(string tipo, string nombre_archivo, int registros_procesados)
        {
            try
            {
                string sql = string.Format("insert into movimientos_tarjetas_archivos (tipo, nombre_archivo,fecha_procesamiento,registros_procesados) values ('{0}','{1}','{2}',{3})",
                    tipo,
                    nombre_archivo,
                    DateTime.Now.ToString("yyyyMMdd HH:mm"),
                    registros_procesados);

                SqlHelper sqlHelp = new SqlHelper(_connString);
                sqlHelp.ExecuteNonQuery(sql);

                Console.WriteLine(sql);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }


        public bool ExisteArchivo(string nombre_archivo)
        {
            try
            {
                string sql = string.Format("select count(*) as contador from movimientos_tarjetas_archivos where tipo='TE' and nombre_archivo = '{0}'",
                    nombre_archivo);

                SqlHelper sqlHelp = new SqlHelper(_connString);
                if (sqlHelp.ExecuteScalar(sql) > 0)
                    return true;
                else
                    return false;

            }
            catch (Exception ex)
            {
                throw ex;
            }
        }


    }
}

